File: Programmers/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer;
        ArrayList<Integer> ans = new ArrayList<Integer>();
        int max_day = 0;
        for(int i = 0; i < progresses.length; i++){
            int day = (100 - progresses[i] + speeds[i] - 1) / speeds[i];
           if(max_day < day){
               ans.add(1);
               max_day = day;
           } 
            else{
                ans.set(ans.size() - 1, ans.get(ans.size() - 1) + 1);
            }   
        }
        answer = new int[ans.size()];
        for(int i = 0; i < ans.size(); i++) answer[i] = ans.get(i);
        return answer;
    }
}

File: Programmers/기능개발.cpp
//https://programmers.co.kr/learn/courses/30/lessons/42586?language=cpp

#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int max_day = 0;
    for(int i = 0; i < speeds.size(); i++){
        int day = (((100 - progresses[i]) + speeds[i] - 1) / speeds[i]);
        if(max_day < day){
            answer.push_back(1);
            max_day = day;
        }
        else {
            answer.back()++;
        }
    }  
    
    return answer;
}